src/coreclr/tools/Common/TypeSystem/IL/Stubs/PInvokeILEmitter.cs

```diff
@@ -300,8 +300,19 @@ private void EmitPInvokeCall(PInvokeILCodeStreams ilCodeStreams)
 
             if (!_pInvokeILEmitterConfiguration.GenerateDirectCall(_targetMethod, out _))
             {
+                int signatureBytes = -1;
+                if (context.Target.IsWindows && context.Target.Architecture == TargetArchitecture.X86 &&
+                    (_targetMethod.GetPInvokeMethodCallingConventions() & UnmanagedCallingConventions.CallingConventionMask) == UnmanagedCallingConventions.Stdcall)
+                {
+                    signatureBytes = 0;
+                    foreach (var p in nativeParameterTypes)
+                    {
+                        signatureBytes += AlignmentHelper.AlignUp(p.GetElementSize().AsInt, context.Target.PointerSize);
+                    }
+                }
+
                 MetadataType lazyHelperType = context.GetHelperType("InteropHelpers");
-                FieldDesc lazyDispatchCell = _interopStateManager.GetPInvokeLazyFixupField(_targetMethod);
+                FieldDesc lazyDispatchCell = _interopStateManager.GetPInvokeLazyFixupField(_targetMethod, signatureBytes);
```

Comment:
I think it would look better to pass in the `nativeSig` to `GetPInvokeLazyFixupField` here, and walk it in the `PInvokeMethodData` constructor to compute the bytes.

Reply:
Yeah, I guess. I had issue with passing the extra parameter which is why initially abandoned that strategy. I'll update it.

Reply:
Done